signer/src/bitcoin/mod.rs

```diff
@@ -82,4 +85,23 @@ pub trait BitcoinInteract: Sync + Send {
         &self,
         txid: &Txid,
     ) -> impl Future<Output = Result<Vec<Txid>, Error>> + Send;
+
+    /// Gets the output of the specified transaction, optionally including
+    /// transactions from the mempool.
+    fn get_transaction_output(
+        &self,
+        outpoint: &bitcoin::OutPoint,
+        include_mempool: bool,
+    ) -> impl Future<Output = Result<Option<GetTxOutResult>, Error>> + Send;
+
+    /// Gets the associated fees for the given transaction. It is expected that
+    /// the provided transaction is known to the Bitcoin core node, either
+    /// confirmed or in the mempool, otherwise an error will be returned.
+    fn get_transaction_fee(&self, tx: &Txid) -> impl Future<Output = Result<Fees, Error>> + Send;
```

Comment:
Note that it looks like the return signature here looks like it's going to change as I'll also need vsize